packages/screen-size.ts

```diff
@@ -0,0 +1,183 @@
+
+  private measure: FromScreenSize;
+
+  private validation(measure: OptionalFromScreenSize) {
+    const keys = Object.keys(measure);
+    const validKeys = [
+      "pt",
+      "px",
+      "in",
+      "cm",
+      "mm",
+      "pc",
+      "picas",
+      "points",
+      "pixels",
+      "inches",
+      "centimeters",
+      "millimeters",
+    ];
+
+    keys.forEach((key) => {
+      if (!validKeys.includes(key)) {
+        throw new Error(`Invalid key: ${key}`);
+      }
+
+      if (typeof measure[key as keyof OptionalFromScreenSize] !== "number") {
+        throw new Error(`Invalid value for ${key}: must be a number`);
+      }
+
+      if (measure[key as keyof OptionalFromScreenSize]! < 0) {
+        throw new Error(`Invalid value for ${key}: cannot be negative`);
+      }
+    });
+
+    if (keys.length === 0) {
+      throw new Error("At least one unit must be specified.");
+    }
+  }
+
+  private constructor(measure: OptionalFromScreenSize) {
+    this.validation(measure);
+    this.measure = {
+      pc: measure.pc ?? 0,
+      pt: measure.pt ?? 0,
+      px: measure.px ?? 0,
+      in: measure.in ?? 0,
+      cm: measure.cm ?? 0,
+      mm: measure.mm ?? 0,
+      picas: measure.picas ?? 0,
+      pixels: measure.pixels ?? 0,
+      points: measure.points ?? 0,
+      inches: measure.inches ?? 0,
+      millimeters: measure.millimeters ?? 0,
+      centimeters: measure.centimeters ?? 0,
+    };
+  }
+
+  static from(measure: OptionalFromScreenSize) {
+    return new ScreenSize(measure);
+  }
+
+  static fromPoints(value: number): ScreenSize {
+    return new ScreenSize({ pt: value });
+  }
+
+  static fromPixels(value: number): ScreenSize {
+    return new ScreenSize({ px: value });
+  }
+
+  static fromInches(value: number): ScreenSize {
+    return new ScreenSize({ in: value });
+  }
+
+  static fromCentimeters(value: number): ScreenSize {
+    return new ScreenSize({ cm: value });
+  }
+
+  static fromMillimeters(value: number): ScreenSize {
+    return new ScreenSize({ mm: value });
+  }
+
+  static fromPicas(value: number): ScreenSize {
+    return new ScreenSize({ pc: value });
+  }
+
+  private convert(
+    value: number,
+    fromUnit: keyof FromScreenSize,
+    toUnit: keyof FromScreenSize
+  ): number {
+    const conversionRates: Partial<Record<keyof FromScreenSize, number>> = {
+      in: 1,
+      inches: 1,
+      mm: ScreenSize.MM_PER_INCH,
+      cm: ScreenSize.CM_PER_INCH,
+      pc: ScreenSize.PICAS_PER_INCH,
+      pt: ScreenSize.POINTS_PER_INCH,
+      px: ScreenSize.PIXELS_PER_INCH,
+      picas: ScreenSize.PICAS_PER_INCH,
+      pixels: ScreenSize.PIXELS_PER_INCH,
+      points: ScreenSize.POINTS_PER_INCH,
+      millimeters: ScreenSize.MM_PER_INCH,
+      centimeters: ScreenSize.CM_PER_INCH,
+    };
+
+    const fromRate = conversionRates[fromUnit] as number;
+    const toRate = conversionRates[toUnit] as number;
+
+    if (!fromRate) {
+      throw new Error(`Invalid 'from' unit: ${fromUnit}`);
```

Comment:
Instead of using a generic Error, consider creating a custom error class like InvalidUnitError. This would make it easier to identify and handle this specific type of error.